### ISSUE TITLE
Improve unit test coverage

### ISSUE THREAD

**Issue by Gudahtt** [MetaMask/eth-hd-keyring#56]
We should strive for 100% unit test coverage, and to cover all input boundary conditions.

**Comment by adonesky1** [comment]
@Gudahtt should we migrate the test suite to Jest so we can get test coverage reporting?

**Comment by mcmire** [comment]
@adonesky1 Yes, we should definitely do this! I don't think there's any reason we shouldn't be using Jest now.

**Comment by adonesky1** [comment]
I'm going to update to use Jest cc @mcmire @Gudahtt

**Comment by adonesky1** [comment]
Current coverage:
![Screen Shot 2022-04-15 at 12 07 49 PM](https://user-images.githubusercontent.com/34557516/163599866-94421e7f-729e-4133-943c-a5a9cbf3fd8d.png)
